fk_reference_objects_commit_meta_txns issue on file deletion

```
0box-sdk        [INFO]   2022/08/12 14:44:01.664225 deleteworker.go:64: https://demo.0chain.net/blobber01 /-scan.png deleted.
0box-sdk        [INFO]   2022/08/12 14:44:01.674448 deleteworker.go:64: https://demo.0chain.net/blobber04 /-scan.png deleted.
0box-sdk        [INFO]   2022/08/12 14:44:01.688512 deleteworker.go:64: https://demo.0chain.net/blobber02 /-scan.png deleted.
0box-sdk        [INFO]   2022/08/12 14:44:02.530968 deleteworker.go:64: https://demo.0chain.net/blobber05 /-scan.png deleted.
0box-sdk        [INFO]   2022/08/12 14:44:10.885672 commitworker.go:102: received a commit request
0box-sdk        [INFO]   2022/08/12 14:44:10.885812 commitworker.go:102: received a commit request
0box-sdk        [INFO]   2022/08/12 14:44:10.885737 commitworker.go:102: received a commit request
0box-sdk        [INFO]   2022/08/12 14:44:10.885742 commitworker.go:102: received a commit request
0box-sdk        [INFO]   2022/08/12 14:44:11.357671 commitworker.go:243: Committing to blobber.https://demo.0chain.net/blobber04
0box-sdk        [INFO]   2022/08/12 14:44:11.628297 commitworker.go:243: Committing to blobber.https://demo.0chain.net/blobber02
0box-sdk        [INFO]   2022/08/12 14:44:12.244439 commitworker.go:243: Committing to blobber.https://demo.0chain.net/blobber05
0box-sdk        [INFO]   2022/08/12 14:44:12.368588 commitworker.go:243: Committing to blobber.https://demo.0chain.net/blobber01
0box-sdk        [ERROR]  2022/08/12 14:44:12.715416 commitworker.go:253: Commit response: 400
0box-sdk        [ERROR]  2022/08/12 14:44:12.715502 commitworker.go:262: https://demo.0chain.net/blobber02 Commit response:{"error":"file_store_error: Error committing to file store. ERROR: update or delete on table \"reference_objects\" violates foreign key constraint \"fk_reference_objects_commit_meta_txns\" on table \"commit_meta_txns\" (SQLSTATE 23503)"}

0box-sdk        [ERROR]  2022/08/12 14:44:12.739095 commitworker.go:253: Commit response: 400
0box-sdk        [ERROR]  2022/08/12 14:44:12.739295 commitworker.go:262: https://demo.0chain.net/blobber05 Commit response:{"error":"file_store_error: Error committing to file store. ERROR: update or delete on table \"reference_objects\" violates foreign key constraint \"fk_reference_objects_commit_meta_txns\" on table \"commit_meta_txns\" (SQLSTATE 23503)"}

0box-sdk        [ERROR]  2022/08/12 14:44:12.886269 commitworker.go:253: Commit response: 400
0box-sdk        [ERROR]  2022/08/12 14:44:12.886486 commitworker.go:262: https://demo.0chain.net/blobber01 Commit response:{"error":"file_store_error: Error committing to file store. ERROR: update or delete on table \"reference_objects\" violates foreign key constraint \"fk_reference_objects_commit_meta_txns\" on table \"commit_meta_txns\" (SQLSTATE 23503)"}

0box-sdk        [ERROR]  2022/08/12 14:44:13.202721 commitworker.go:253: Commit response: 400
0box-sdk        [ERROR]  2022/08/12 14:44:13.202811 commitworker.go:262: https://demo.0chain.net/blobber04 Commit response:{"error":"file_store_error: Error committing to file store. ERROR: update or delete on table \"reference_objects\" violates foreign key constraint \"fk_reference_objects_commit_meta_txns\" on table \"commit_meta_txns\" (SQLSTATE 23503)"}

0box-sdk        [INFO]   2022/08/12 14:44:13.202830 deleteworker.go:178: Commit failedhttps://demo.0chain.net/blobber05commit_error: {"error":"file_store_error: Error committing to file store. ERROR: update or delete on table \"reference_objects\" violates foreign key constraint \"fk_reference_objects_commit_meta_txns\" on table \"commit_meta_txns\" (SQLSTATE 23503)"}
0box-sdk        [INFO]   2022/08/12 14:44:13.202841 deleteworker.go:178: Commit failedhttps://demo.0chain.net/blobber01commit_error: {"error":"file_store_error: Error committing to file store. ERROR: update or delete on table \"reference_objects\" violates foreign key constraint \"fk_reference_objects_commit_meta_txns\" on table \"commit_meta_txns\" (SQLSTATE 23503)"}
0box-sdk        [INFO]   2022/08/12 14:44:13.202851 deleteworker.go:178: Commit failedhttps://demo.0chain.net/blobber02commit_error: {"error":"file_store_error: Error committing to file store. ERROR: update or delete on table \"reference_objects\" violates foreign key constraint \"fk_reference_objects_commit_meta_txns\" on table \"commit_meta_txns\" (SQLSTATE 23503)"}
0box-sdk        [INFO]   2022/08/12 14:44:13.202892 deleteworker.go:178: Commit failedhttps://demo.0chain.net/blobber04commit_error: {"error":"file_store_error: Error committing to file store. ERROR: update or delete on table \"reference_objects\" violates foreign key constraint \"fk_reference_objects_commit_meta_txns\" on table \"commit_meta_txns\" (SQLSTATE 23503)"}
Delete failed. Delete failed: Commit consensus failed
```

--- a/code/go/0chain.net/blobbercore/allocation/deletefilechange.go
+++ b/code/go/0chain.net/blobbercore/allocation/deletefilechange.go
@@ -63,6 +63,7 @@ func (nf *DeleteFileChange) DeleteTempFile() error {
 func (nf *DeleteFileChange) CommitToFileStore(ctx context.Context) error {
 	db := datastore.GetStore().GetTransaction(ctx)
 	type Result struct {
+		Id            string
 		ContentHash   string
 		ThumbnailHash string
 	}
@@ -71,7 +72,7 @@ func (nf *DeleteFileChange) CommitToFileStore(ctx context.Context) error {
 	wg := &sync.WaitGroup{}
 	var results []Result
 	err := db.Model(&reference.Ref{}).Unscoped().
-		Select("content_hash", "thumbnail_hash").
+		Select("id", "content_hash", "thumbnail_hash").
 		Where("allocation_id=? AND path LIKE ? AND type=? AND deleted_at is not NULL",
 			nf.AllocationID, nf.Path+"%", reference.FILE).
 		FindInBatches(&results, 100, func(tx *gorm.DB, batch int) error {
@@ -81,6 +82,8 @@ func (nf *DeleteFileChange) CommitToFileStore(ctx context.Context) error {
 				tx.Model(&reference.Ref{}).
 					Where("allocation_id=? AND content_hash=?", nf.AllocationID, res.ContentHash).
 					Count(&count)
+
+				tx.Model(&reference.CommitMetaTxn{}).Unscoped().Delete("ref_id = ?", res.Id)
 
 				if count != 0 && res.ThumbnailHash == "" {
 					continue

--- a/code/go/0chain.net/blobbercore/allocation/deletefilechange.go
+++ b/code/go/0chain.net/blobbercore/allocation/deletefilechange.go
@@ -83,7 +83,7 @@ func (nf *DeleteFileChange) CommitToFileStore(ctx context.Context) error {
 					Where("allocation_id=? AND content_hash=?", nf.AllocationID, res.ContentHash).
 					Count(&count)
 
-				tx.Model(&reference.CommitMetaTxn{}).Unscoped().Delete("ref_id = ?", res.Id)
+				db.Model(&reference.CommitMetaTxn{}).Delete("ref_id = ?", res.Id)
 
 				if count != 0 && res.ThumbnailHash == "" {
 					continue

